pennylane/ops/op_math/prod.py

```diff
@@ -366,3 +366,55 @@ def simplify(self) -> Union["Prod", Sum]:
         factors = [Prod(*factor).simplify() if len(factor) > 1 else factor[0] for factor in factors]
 
         return Sum(*factors)
+
+
+def _prod_sort(op_list, wire_map: dict = None):
+    """Insertion sort algorithm that sorts a list of product factors by their wire indices, taking
+    into account the operator commutivity.
```

Comment:
Since we are only considering wire intersections and not full commutativity, maybe we should specify as such in the documentation.  

Or we could use `is_commuting` instead of `intersect1d`.  But that might add more computational complexity than we want.

Reply:
Changed docstring.

For now this does the trick. If we need better sorting we should improve `is_commuting` to support `Prod` and `Sum` operators, and then we could use it here.